tools/nd_freeze

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+#
+# Install access to the NeuroDebian snapshot repository that allows access
+# to old packages based on dates and version numbers.
+#
+# The date/time format sent to NeuroDebian to get the repo info is in the
+# format yyyymmddThhmmssz or yyymmdd. However, the script will handle and
+# reformat a number of date/time formats. Quotes are required if there is a
+# space between the date and time components:
+#
+#   yyyymmddThh:mm:ssZ
+#   yyyymmdd
+#   mm/dd/yyyy
+#   yyyy-mm-dd
+#   yyyymmddThh:mm
+#   "yyyy-mm-dd hh:mm:ss"
+#   "mm/dd/yy hh:mm:ss"
+#
+# Browse available snapshots here:
+# http://snapshot-neuro.debian.net:5002/archive/neurodebian/
+#
+# If there is no import at the exact time you specified you will get the latest
+# available timestamp which is before the time you specified.
+
+set -e
+APT_SOURCE_PATH='/etc/apt/sources.list.d/neurodebian-snapshots.sources.list'
+
+
+# Check for date command line argument
+if [ -z "$1" ]; then
+cat << EOT
+Script to enable a NeurDebian archived snapshot repository. Snapshot repositories
+provide previous releases of packages based on point-of-time archiving.
+
+See: http://snapshot-neuro.debian.net:5002/ 
+
+Synopsis
+--------
+
+  nd_freeze <date>
+
+  Valid date formats include:
+    yyyymmddThh:mm:ssZ
+    yyyymmdd
+    mm/dd/yyyy
+    yyyy-mm-dd
+    yyyymmddThh:mm
+    "yyyy-mm-dd hh:mm:ss"
+    "mm/dd/yy hh:mm:ss"
+
+EOT
+exit 1
+fi
+user_date=$1
+
+
+# Check for existence of required command line utilities
+programs='wget sed'
```

Comment:
Please make it possible to use curl instead of wget (ideally could try both but I guess would be trickier so probably can just switch to use curl).  I found curl more frequently present than wget, e.g. in current stock Debian:latest docker image.
I will push a fix for that since need to get it going

Reply:
eh that sucks:
```shell
$> docker run -it --rm -v ~/dicom-test:/dicoms --entrypoint=bash debian:sid
root@79268d5ebfef:/# wget
bash: wget: command not found
root@79268d5ebfef:/# curl
bash: curl: command not found
```
I wonder if there is another "internet" aware utility/thing for basic http transaction likely to reside on such basic/stripped instance...  there is perl...

Reply:
ok, here is my patch for `curl` just in case 
```patch
$> git diff
diff --git a/tools/nd_freeze b/tools/nd_freeze
index cd5b7ec..100b0b7 100755
--- a/tools/nd_freeze
+++ b/tools/nd_freeze
@@ -55,7 +55,7 @@ user_date=$1
 
 
 # Check for existence of required command line utilities
-programs='wget sed'
+programs='curl sed'
 for program in $programs
 do
   if [ -z `which ${program}` ]; then
@@ -133,12 +133,12 @@ fi
 
 # Knock on snapshot's door. This is temporarily necessary until the production
 # version of the site is made available.
-wget -O /dev/null -q http://neuro.debian.net/_files/knock-snapshots
+curl -s http://neuro.debian.net/_files/knock-snapshots >| /dev/null
 
 
 # Add "next" snapshot to cover package updates made after the previous snapshot
 # but before the user provided date.
-next_line=`wget -qO- http://snapshot-neuro.debian.net:5002/archive/neurodebian/${previous_timestamp}/ | sed -n '/>next<\/a>/p'`
+next_line=`curl -s http://snapshot-neuro.debian.net:5002/archive/neurodebian/${previous_timestamp}/ | sed -n '/>next<\/a>/p'`
 next_timestamp=`echo "$next_line" | sed -E 's/^.*\/archive\/neurodebian\/([0-9TZ]+)\/">next.*$/\1/'`
 if [ -z "$next_timestamp" ]; then
```
to install `curl` in the stock docker image ATM I need
```
5 upgraded, 26 newly installed, 0 to remove and 82 not upgraded.
Need to get 10.5 MB of archives.
After this operation, 18.3 MB of additional disk space will be used.
```
to install `wget`
```
2 upgraded, 11 newly installed, 0 to remove and 85 not upgraded.
Need to get 6956 kB of archives.
After this operation, 12.4 MB of additional disk space will be used.
```
so, fixes should

- [ ] provide support for either having `wget` or `curl` installed
- [ ] if none of those two is available, should `apt-get install  --no-install-recommends wget` since altogether seems to be leaner.  

NB since perl is there, there might be a way or just a single package to install to get some basic "fetch from the web" functionality working... just dunno one yet